app/controllers/users_controller.rb

```diff
@@ -171,6 +171,11 @@ def show
 		if current_user.recipes.blank?
 			@calories = self.class.calc_calories(@gender, @weight, @height, @age, @exercise, @goal).round(0)
 			@all_recipes = Recipe.find_in_api(@calories, @budget, @time)
+			if @all_recipes.nil?
+				flash[:error] = "401: Request Denied"
```

Comment:
Good catch: the API always returns a 401 error code when it has reached its limit. Maybe it would be better to put a descriptive message, such as "API limit reached, please try again in one minute!"

Reply:
Agreed! This is definitely better than having the error just arise and confuse the user. Also agreed with Omid A that we should have a more descriptive & user friendly alert caption.